### PR TITLE
Pan camera in editor 2

### LINKED ISSUE #1690: Add panning control to the editor camera
not sure how it should be controlled but it should be made possible for the player to pan the camera around in the editor

### PR DESCRIPTION
**Pull request by FuJa0815**
Replaces #1697
Closes #1690

You can move the cam using your arrow keys or using your middle mouse button.
Using the spacebar, you can reset the cam to the default.
There is currently no limitation in how far away you can pan your camera.

I was not able to test if MMB works, because mine is broken. RMB and a key works.

### CODE REVIEW

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -543,6 +549,45 @@ public void CreateNewMicrobe()
         EnqueueAction(action);
     }
 
+    [RunOnAxisGroup]
+    [RunOnAxis(new[] { "e_pan_up", "e_pan_down" }, new[] { -1.0f, 1.0f })]
+    [RunOnAxis(new[] { "e_pan_left", "e_pan_right" }, new[] { -1.0f, 1.0f })]
+    public void PanCamWithKeys(float delta, float upDown, float leftRight)
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:555]: `PanCameraWithKeys`

File: src/engine/input/RunOnKeyUpAttribute.cs [hunk 2]
```diff
@@ -12,8 +12,7 @@ public RunOnKeyUpAttribute(string inputName) : base(inputName)
 
     public override bool OnInput(InputEvent @event)
     {
-        var before = HeldDown;
-        if (!base.OnInput(@event) && before && !HeldDown)
+        if (base.OnInput(@event) && !HeldDown)
```
Comment (hhyyrylainen) [inline src/engine/input/RunOnKeyUpAttribute.cs:15]: The PR making just this change needs accepting before this PR can be merged...

File: project.godot [hunk 3]
```diff
@@ -203,6 +203,36 @@ help={
...
+"deadzone": 0.5,
+"events": [ Object(InputEventKey,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":false,"shift":false,"control":false,"meta":false,"command":false,"pressed":false,"scancode":16777234,"unicode":0,"echo":false,"script":null)
+ ]
+}
+e_pan_mouse={
+"deadzone": 0.5,
+"events": [ Object(InputEventMouseButton,"resource_local_to_scene":false,"resource_name":"","device":0,"alt":false,"shift":false,"control":false,"meta":false,"command":false,"button_mask":0,"position":Vector2( 0, 0 ),"global_position":Vector2( 0, 0 ),"factor":1.0,"button_index":3,"pressed":false,"doubleclick":false,"script":null)
+ ]
+}
+e_reset_cam={
```
Comment (hhyyrylainen) [inline project.godot:231]: `e_reset_camera`

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 4]
```diff
@@ -830,6 +875,15 @@ internal void ChangeMutationPoints(int change)
         MutationPoints = (MutationPoints + change).Clamp(0, Constants.BASE_MUTATION_POINTS);
     }
 
+    /// <summary>
+    ///   Called by PlayerMicrobeEditorInput
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:879]: Is this comment accurate? This looks to be called from within this class. And summary should be what this does, not who calls this.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 5]
```diff
@@ -830,6 +875,15 @@ internal void ChangeMutationPoints(int change)
...
+    /// </summary>
+    /// <param name="vector">The direction to move the camera</param>
+    private void MoveObjectToFollow(Vector3 vector)
+    {
+        camera.ObjectToFollow.Translation += vector;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:884]: This should use `cameraFollow` and not go through the camera object.

File: src/microbe_stage/MicrobeCamera.cs [hunk 6]
```diff
@@ -161,25 +166,32 @@ public void Zoom(float delta, float value)
     /// </summary>
     public override void _PhysicsProcess(float delta)
     {
+        var currFloorPos = new Vector3(Translation.x, 0, Translation.z);
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeCamera.cs:169]: `curr` -> `current` everywhere in this method.

File: src/microbe_stage/MicrobeCamera.cs [hunk 7]
```diff
@@ -161,25 +166,32 @@ public void Zoom(float delta, float value)
...
+        var currCamHeight = new Vector3(0, Translation.y, 0);
+        var newCamPos = new Vector3(0, CameraHeight, 0);
         if (ObjectToFollow != null)
         {
-            var target = ObjectToFollow.Transform.origin + new Vector3(0, CameraHeight, 0);
+            var newFloorPos = new Vector3(ObjectToFollow.Transform.origin.x, 0, ObjectToFollow.Transform.origin.z);
```
Comment (hhyyrylainen) [inline src/microbe_stage/MicrobeCamera.cs:174]: `pos` -> `position`

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: locale/bg.po [hunk 8]
```diff
@@ -2,12 +2,12 @@
 # Copyright (C) 2020 ORGANIZATION
 # This file is distributed under the same license as the PROJECT project.
 # FIRST AUTHOR <EMAIL@ADDRESS>, 2020.
-#
+#
```
Comment (hhyyrylainen) [inline locale/bg.po:5]: What's with these blank spaces getting added?

Comment (FuJa0815) [reply locale/bg.po:5]: strange... I'll revert them manually

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**